allennlp/fairness/fairness_metrics.py

```diff
@@ -0,0 +1,921 @@
+"""
+Fairness metrics are based on:
+
+1. Barocas, S.; Hardt, M.; and Narayanan, A. 2019. [Fairness and machine learning](https://fairmlbook.org).
+
+2. Zhang, B. H.; Lemoine, B.; and Mitchell, M. 2018. [Mitigating unwanted biases with adversarial learning]
+(https://api.semanticscholar.org/CorpusID:9424845).
+In Proceedings of the 2018 AAAI/ACM Conference on AI, Ethics, and Society, 335-340.
+
+3. Hardt, M.; Price, E.; Srebro, N.; et al. 2016. [Equality of opportunity in supervised learning]
+(https://api.semanticscholar.org/CorpusID:7567061). In Advances in Neural Information Processing Systems,
+3315–3323.
+
+4. Beutel, A.; Chen, J.; Zhao, Z.; and Chi, E. H. 2017. [Data decisions and theoretical implications when
+adversarially learning fair representations](https://api.semanticscholar.org/CorpusID:24990444).
+arXiv preprint arXiv:1707.00075.
+
+5. Aka, O.; Burke, K.; Bäuerle, A.; Greer, C.; and Mitchell, M. 2021.
+[Measuring model biases in the absence of ground truth](https://api.semanticscholar.org/CorpusID:232135043).
+arXiv preprint arXiv:2103.03417.
+
+It is provably impossible to satisfy any two of Independence, Separation, and Sufficiency simultaneously,
+except in degenerate cases.
```

Comment:
That's interesting. Do you have a reference for that?

Reply:
Added a link in the doctoring!